### PR TITLE
ADEN-3609 SlotTweaker animations API

### PR DESCRIPTION
**Pull request by fraszczakszymon**
Possibility to animate slots using postMessage

### CODE REVIEW

File: extensions/wikia/AdEngine/js/SlotTweaker.js [hunk 1]
```diff
@@ -179,6 +182,47 @@ define('ext.wikia.adEngine.slotTweaker', [
...
+	function expand(slotName) {
+		var slot = doc.getElementById(slotName);
+
+		slot.style.maxHeight = slot.offsetHeight + 'px';
+		DOMElementTweaker.removeClass(slot, 'hidden');
+		DOMElementTweaker.addClass(slot, 'slot-animation');
+		slot.style.maxHeight = slot.scrollHeight + 'px';
+	}
+
+	function messageCallback(data) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/SlotTweaker.js]: MAJOR Function has a complexity of 13 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)

File: extensions/wikia/AdEngine/js/MessageListener.js [hunk 2]
```diff
@@ -55,7 +55,9 @@ define('ext.wikia.adEngine.messageListener', [
 					log(['Event matching', msg, callback], 'debug', logGroup);
 
 					callback.fn(JSON.parse(msg.data).AdEngine);
-					callbacks.splice(i, 1);
+					if (!callback.match.infinite) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/MessageListener.js]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)

File: extensions/wikia/AdEngine/js/run/mercury.run.js [hunk 3]
```diff
@@ -9,25 +9,29 @@ require([
...
-			 rubiconVulcan,
-			 customAdsLoader,
-			 messageListener,
-			 mercuryListener,
-			 scrollHandler,
-			 yavliTag,
-			 geo,
-			 instantGlobals,
-			 win) {
+], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/run/mercury.run.js:18]: MAJOR Function has 15 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* CRITICAL 3 critical
* MAJOR 6 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [MessageListener.js#L27](https://github.com/Wikia/app/blob/c0e7ee61048ce9c110da5b1e9a4726ffb113e3fd/extensions/wikia/AdEngine/js/MessageListener.js#L27): Change this condition so that it does not always evaluate to "true". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2583)
1. CRITICAL [SlotTweaker.js#L237](https://github.com/Wikia/app/blob/c0e7ee61048ce9c110da5b1e9a4726ffb113e3fd/extensions/wikia/AdEngine/js/SlotTweaker.js#L237): "noop" expects "0" arguments, but "1" were provided. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS930)
1. MAJOR [desktop.run.js#L23](https://github.com/Wikia/app/blob/c0e7ee61048ce9c110da5b1e9a4726ffb113e3fd/extensions/wikia/AdEngine/js/run/desktop.run.js#L23): Function has 19 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [desktop.run.js#L115](https://github.com/Wikia/app/blob/c0e7ee61048ce9c110da5b1e9a4726ffb113e3fd/extensions/wikia/AdEngine/js/run/desktop.run.js#L115): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [SlotTweaker.js#L74](https://github.com/Wikia/app/blob/c0e7ee61048ce9c110da5b1e9a4726ffb113e3fd/extensions/wikia/AdEngine/js/SlotTweaker.js#L74): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
1. INFO [SlotTweaker.js#L82](https://github.com/Wikia/app/blob/c0e7ee61048ce9c110da5b1e9a4726ffb113e3fd/extensions/wikia/AdEngine/js/SlotTweaker.js#L82): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

File: extensions/wikia/AdEngine/js/MessageListener.js (hunk 2 above)
Comment (drets) [inline extensions/wikia/AdEngine/js/MessageListener.js]: _Blocks are nested too deeply. (4) [W073]_

Consider moving few `if`s to a separate function.

File: extensions/wikia/AdEngine/js/SlotTweaker.js [hunk 4]
```diff
@@ -179,6 +182,47 @@ define('ext.wikia.adEngine.slotTweaker', [
 		return;
 	}
 
+	function collapse(slotName) {
+		var slot = doc.getElementById(slotName);
+
+		slot.style.maxHeight = slot.scrollHeight + 'px';
+		noop(slot.offsetHeight);
```
Comment (drets) [inline extensions/wikia/AdEngine/js/SlotTweaker.js]: Please, give `noop` meaningful name.

File: extensions/wikia/AdEngine/js/SlotTweaker.js [hunk 5]
```diff
@@ -179,6 +182,47 @@ define('ext.wikia.adEngine.slotTweaker', [
...
+		DOMElementTweaker.addClass(slot, 'slot-animation');
+		slot.style.maxHeight = slot.scrollHeight + 'px';
+	}
+
+	function messageCallback(data) {
+		if (!data.slotName) {
+			return log('messageCallback: missing slot name', log.levels.debug, logGroup);
+		}
+
+		switch (data.action) {
```
Comment (drets) [inline extensions/wikia/AdEngine/js/SlotTweaker.js:204]: Agree with Sonar, log default case.

File: extensions/wikia/AdEngine/js/SlotTweaker.js (hunk 1 above)
Comment (drets) [inline extensions/wikia/AdEngine/js/SlotTweaker.js]: Consider:
```
	function messageCallback(data) {
		if (!data.slotName) {
			log('messageCallback: missing slot name', log.levels.debug, logGroup);
			return;
		}

		switch (data.action) {
			case 'expand':
				expand(data.slotName);
				break;
			case 'collapse':
				collapse(data.slotName);
				break;
			case 'hide':
				hide(data.slotName);
				break;
			case 'show':
			        show(data.slotName);
				break;
			case 'make-responsive':
				makeResponsive(data.slotName, data.aspectRatio);
				break;
...
```

**Comment by WikiaTech** [comment]
SonarQube analysis reported 6 issues
* MAJOR 4 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [desktop.run.js#L23](https://github.com/Wikia/app/blob/2927a807204467dc082a7a1210977989885472a6/extensions/wikia/AdEngine/js/run/desktop.run.js#L23): Function has 19 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [desktop.run.js#L115](https://github.com/Wikia/app/blob/2927a807204467dc082a7a1210977989885472a6/extensions/wikia/AdEngine/js/run/desktop.run.js#L115): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [SlotTweaker.js#L74](https://github.com/Wikia/app/blob/2927a807204467dc082a7a1210977989885472a6/extensions/wikia/AdEngine/js/SlotTweaker.js#L74): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
1. INFO [SlotTweaker.js#L82](https://github.com/Wikia/app/blob/2927a807204467dc082a7a1210977989885472a6/extensions/wikia/AdEngine/js/SlotTweaker.js#L82): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [desktop.run.js#L23](https://github.com/Wikia/app/blob/abc59010ff4b99f20fbbfc19e2cf7fdd79dbd52b/extensions/wikia/AdEngine/js/run/desktop.run.js#L23): Function has 19 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [desktop.run.js#L115](https://github.com/Wikia/app/blob/abc59010ff4b99f20fbbfc19e2cf7fdd79dbd52b/extensions/wikia/AdEngine/js/run/desktop.run.js#L115): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [SlotTweaker.js#L74](https://github.com/Wikia/app/blob/abc59010ff4b99f20fbbfc19e2cf7fdd79dbd52b/extensions/wikia/AdEngine/js/SlotTweaker.js#L74): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
1. INFO [SlotTweaker.js#L82](https://github.com/Wikia/app/blob/abc59010ff4b99f20fbbfc19e2cf7fdd79dbd52b/extensions/wikia/AdEngine/js/SlotTweaker.js#L82): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* MAJOR 3 major
* INFO 2 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [desktop.run.js#L23](https://github.com/Wikia/app/blob/26931dc5c1e819f229a23c9099738f17c7a7cfeb/extensions/wikia/AdEngine/js/run/desktop.run.js#L23): Function has 19 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [desktop.run.js#L115](https://github.com/Wikia/app/blob/26931dc5c1e819f229a23c9099738f17c7a7cfeb/extensions/wikia/AdEngine/js/run/desktop.run.js#L115): Function has 11 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. INFO [SlotTweaker.js#L74](https://github.com/Wikia/app/blob/26931dc5c1e819f229a23c9099738f17c7a7cfeb/extensions/wikia/AdEngine/js/SlotTweaker.js#L74): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)
1. INFO [SlotTweaker.js#L82](https://github.com/Wikia/app/blob/26931dc5c1e819f229a23c9099738f17c7a7cfeb/extensions/wikia/AdEngine/js/SlotTweaker.js#L82): Complete the task associated to this TODO comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)

**Review by drets — APPROVED** [review]

**Pull request closed without merging**